source/languageHandler.py

```diff
@@ -213,76 +319,92 @@ def setLanguage(lang: str) -> None:
-
-	if '-' in localeName:
-		# Python couldn't support the language-country locale, try language_country.
+		log.debugWarning(f"Locale {localeName} not supported by Windows")
+	if localeString:
 		try:
-			localeName = localeName.replace('-', '_')
-			locale.setlocale(locale.LC_ALL, localeName)
+			locale.setlocale(locale.LC_ALL, localeString)
 			locale.getlocale()
-			log.debug(f"set python locale to {localeName}")
+			log.debug(f"set python locale to {localeString}")
 			return
 		except locale.Error:
-			log.debugWarning(f"python locale {localeName} could not be set")
+			failedToSetLocale = True
+			log.debugWarning(f"python locale {localeString} could not be set")
 		except ValueError:
-			log.debugWarning(f"python locale {localeName} could not be retrieved with getlocale")
-
-	if '_' in localeName:
-		# Python couldn't support the language_country locale, just try language.
+			failedToSetLocale = True
+			log.debugWarning(f"python locale {localeString} could not be retrieved with getlocale")
+		# The full form langName_country either cannot be retrieved from Windows
+		# or Python cannot be set to that locale.
+		# Try just with the language name.
+		if "_" in localeName:
+			localeName = localeName.split("_")[0]
+			try:
+				localeString = localeStringFromLocaleCode(localeName)
+				log.debug(f"Win32 locale string from locale code is {localeString}")
+			except ValueError:
+				log.debugWarning(f"Locale {localeName} not supported by Windows")
+	if localeString:
 		try:
-			localeName = localeName.split('_')[0]
-			locale.setlocale(locale.LC_ALL, localeName)
+			locale.setlocale(locale.LC_ALL, localeString)
 			locale.getlocale()
-			log.debug(f"set python locale to {localeName}")
+			log.debug(f"set python locale to {localeString}")
 			return
 		except locale.Error:
-			log.debugWarning(f"python locale {localeName} could not be set")
+			failedToSetLocale = True
+			log.debugWarning(f"python locale {localeString} could not be set")
 		except ValueError:
-			log.debugWarning(f"python locale {localeName} could not be retrieved with getlocale")
-
-	try:
-		locale.getlocale()
-	except ValueError:
-		# as the locale may have been changed to something that getlocale() couldn't retrieve
+			failedToSetLocale = True
+			log.debugWarning(f"python locale {localeString} could not be retrieved with getlocale")
+		# As a final fallback try setting locale just to the English name of the given language.
+		localeFromLang = englishLanguageNameFromNVDALocale(localeName)
+		if localeFromLang:
+			try:
+				locale.setlocale(locale.LC_ALL, localeFromLang)
+				locale.getlocale()
+				log.debug(f"set python locale to {localeFromLang}")
+				return
+			except locale.Error:
+				failedToSetLocale = True
+				log.debugWarning(f"python locale {localeFromLang} could not be set")
+			except ValueError:
+				failedToSetLocale = True
+				log.debugWarning(f"python locale {localeFromLang} could not be retrieved with getlocale")

```

Comment:
I think the indentation can be reduced here too

```suggestion
	# As a final fallback try setting locale just to the English name of the given language.
	localeFromLang = englishLanguageNameFromNVDALocale(localeName)
	if localeFromLang:
		try:
			locale.setlocale(locale.LC_ALL, localeFromLang)
			locale.getlocale()
			log.debug(f"set python locale to {localeFromLang}")
			return
		except locale.Error:
			failedToSetLocale = True
			log.debugWarning(f"python locale {localeFromLang} could not be set")
		except ValueError:
			failedToSetLocale = True
			log.debugWarning(f"python locale {localeFromLang} could not be retrieved with getlocale")

```